File: chapter-5/function/calculator.js
x = 5
y = 10

fact = function (n) {
    let ans = 1
    for(let i = 1; i <= n; i++) ans *= i
    return ans
}

console.log(`Summation = ${x} + ${y} = ${x + y}`)
console.log(`Subtraction = ${x} - ${y} = ${x - y}`)
console.log(`Multiplication = ${x} * ${y} = ${x * y}`)
console.log(`Division = ${x} / ${y} = ${x / y}`)
console.log(`Modulus = ${x} % ${y} = ${x % y}`)
console.log(`Exponentiation = ${x} ** ${y} = ${x ** y}`)
console.log(`Logarithm = log${x} = ${Math.log(x)}`)
console.log(`Factorial = ${x}! = ${fact(x)}`)
console.log(`Square root = sqrt(${x}) = ${Math.sqrt(x)}`)
console.log(`Cube root = cbrt(${x}) = ${Math.cbrt(x)}`)
console.log(`Power = pow(${x}, ${y}) = ${Math.pow(x, y)}`)
console.log(`Absolute = abs(${x}) = ${Math.abs(x)}`)
console.log(`Ceil = ceil(${x}) = ${Math.ceil(x)}`)
console.log(`Floor = floor(${x}) = ${Math.floor(x)}`)
console.log(`Round = round(${x}) = ${Math.round(x)}`)
console.log(`Trunc = trunc(${x}) = ${Math.trunc(x)}`)
console.log(`Max = max(${x}, ${y}) = ${Math.max(x, y)}`)
console.log(`Min = min(${x}, ${y}) = ${Math.min(x, y)}`)
console.log(`Sin = sin(${x}) = ${Math.sin(x)}`)
console.log(`Cos = cos(${x}) = ${Math.cos(x)}`)
console.log(`Tan = tan(${x}) = ${Math.tan(x)}`)

File: chapter-5/function/calculator.c
// building a calculator using c programming language
// our calculator will be able to perform the following operations
// addition, subtraction, multiplication, division, modulus, power, factorial, square root, cube root, logarithm, trigonometric functions

#include <stdio.h>
#include <math.h>

int fact(int n){
    int ans = 1;
    // 5! = 5*4*3*2*1 = 120
    for(int i = n; i >= 1; i--){
        ans *= i;
    }
    /*
        i = 5 --> 5 >= 1 --> true --> ans = 1 * 5 = 5
        i = 4 --> 4 >= 1 --> true --> ans = 5 * 4 = 20
        i = 3 --> 3 >= 1 --> true --> ans = 20 * 3 = 60
        i = 2 --> 2 >= 1 --> true --> ans = 60 * 2 = 120
        i = 1 --> 1 >= 1 --> true --> ans = 120 * 1 = 120
        i = 0 --> 0 >= 1 --> false --> exit loop
    */
    return ans;
}
int power(int m, int n){
    // calculate power of m to n
    int ans = 1;
    for(int i = 1; i <= n; i++){
        ans *= m;
    }
    return ans;
}

int main(){
    int a, b, c;
    printf("Enter two numbers: ");
    scanf("%d%d", &a, &b);
    
    printf("Summation: %d + %d = %d\n", a, b, a+b);
    printf("Subtraction: %d - %d = %d\n", a, b, a-b);
    printf("Multiplication: %d * %d = %d\n", a, b, a*b);
    printf("Division: %d / %d = %d\n", a, b, a/b);
    printf("Modulus: %d %% %d = %d\n", a, b, a%b);
    printf("Power: %d ^ %d = %d\n", a, b, power(a, b));
    printf("Factorial: %d! = %d\n", a, fact(a));
    printf("Square root: √(%d) = %f\n", a, sqrt(a));
    printf("Cube root: ∛(%d) = %f\n", a, cbrt(a));
    printf("Logarithm: log(%d) = %f\n", a, log(a));
    printf("Trigonometric functions: sin(%d) = %f, cos(%d) = %f, tan(%d) = %f\n", a, sin(a), a, cos(a), a, tan(a));

    return 0;
}